### ISSUE TITLE
Deprecation warnings with ZMQ 0.3.4

### ISSUE THREAD

**Issue by ViralBShah** [JuliaLang/IJulia.jl#468]
I thought IJulia had gone through a deprecation warnings cleanup, but apparently not:

```
WARNING: bytestring(zmsg::Message) is deprecated, use unsafe_string(zmsg::Message) instead.
while loading /home/viral/.julia/v0.5/IJulia/src/kernel.jl, in expression starting on line 31
```

**Comment by ViralBShah** [comment]
This is on Julia 0.5. I am using a tunneled connection to connect to Jupyter running on arm, but I don't think that has to do with this. Seems like `bytestring` in msg.jl needs to be updated to `unsafe_string`.

**Comment by dlfivefifty** [comment]
I'm getting a couple more

```julia
WARNING: Base.writemime is deprecated.
  likely near /Users/solver/.julia/v0.5/Interact/src/IJulia/setup.jl:116
WARNING: Base.writemime is deprecated.
  likely near /Users/solver/.julia/v0.5/Interact/src/IJulia/setup.jl:116
```

**Comment by stevengj** [comment]
@dlfivefifty, those are deprecations in the Interact package, not in IJulia.

@ViralBShah, the new deprecations are apparently because ZMQ.jl was updated (they are deprecations from the ZMQ package, not from Base.)

**Comment by ViralBShah** [comment]
I thought that the package was free of deprecations, and this explains it. Thanks.

**Comment by yuyichao** [comment]
https://github.com/JuliaLang/IJulia.jl/pull/444

**Comment by wookay** [comment]
rebased, and waiting review. thanks.

**Comment by stevengj** [comment]
Closed by #444.
